### PR TITLE
Fix EZP-25294: do not set the location if there isn't one

### PR DESCRIPTION
**Pull request by bdunogier**
> Fixes http://jira.ez.no/browse/EZP-25294 and https://jira.ez.no/browse/EZP-25470
> Replaces #1542

Fixes a fatal error that occurs when an item  is viewed as an embed. The `location` is set to null instead of not being set at all. This prevents the variable from being set if there is no location, as [suggested by @andrerom](https://github.com/ezsystems/ezpublish-kernel/pull/1542#issuecomment-169713854).

### Backward compatibility
This should be okay. If a location is being viewed, the variable will be set. If a content is being viewed, users do not expect the variable to exist.

Also fixes the solr integration test failure ([EZP-25470](https://jira.ez.no/browse/EZP-25470)) in pull-requests against 6.1.

### CODE REVIEW

**Comment by dpobel** [comment]
+1 (with fixed test)

**Comment by bdunogier** [comment]
As said during the daily today, I think I know why solr is failing.

We require the solr package from kernel, and solr requires kernel `~6.0@dev | 5.4@dev`. It should be satisfied by the 6.1 branch, and that's actually why [travis on the 6.1 branch passes](https://travis-ci.org/ezsystems/ezpublish-kernel/builds/108519887).

But when we're on a _PR based on the 6.1 branch, it seems that the kernel version is either incorrect, or unknown. The solr package's requirements aren't satisfied, and the installation fails.

It seems that there is a composer environment variable that we can use for this, I'll try on this PR.

**Comment by andrerom** [comment]
[The fun of de coupled git repositories](http://www.whitewashing.de/2015/04/11/monolithic_repositories_with_php_and_composer.html#downsides-of-having-many-repositories) :)

**Comment by bdunogier** [comment]
HAH !

https://getcomposer.org/doc/articles/troubleshooting.md#package-not-found-on-travis-ci-org

> the problem might be that Composer is not able to detect the version of the package properly. If it is a git clone it is generally alright and Composer will detect the version of the current branch, but travis does shallow clones so that process can fail

That's why. All we need to do is run `git fetch --unshallow` on the tested branch, and it should all work. Will try.

TL;DR: We had the same issue with the upgrade process, where the merge of a stable tag would generate a ton of conflicts because of the lack of history.

File: bin/.travis/prepare_unittest.sh [hunk 1]
```diff
@@ -12,6 +12,15 @@ if [ "$DB" = "postgresql" ] ; then psql -c "CREATE DATABASE $DB_NAME;" -U postgr
 echo "> Setup github auth key to not reach api limit"
 cp bin/.travis/composer-auth.json ~/.composer/auth.json
 
+# solr package search API integration tests
+if [ "$TEST_CONFIG" = "phpunit-integration-legacy-solr.xml" ] ; then
+    composer require -v --no-progress --no-interaction ezsystems/ezplatform-solr-search-engine:dev-master
```
Comment (andrerom) [inline bin/.travis/prepare_unittest.sh]: Cool, but to avoid issues with this doing composer install for us, can change this further to something like:

```bash
COMPOSER_UPDATE=""
# solr package search API integration tests
if [ "$TEST_CONFIG" = "phpunit-integration-legacy-solr.xml" ] ; then
    composer require --no-update ezsystems/ezplatform-solr-search-engine:dev-master
    COMPOSER_UPDATE="true"

    # required to satisfy the requirements of ezsystems/ezplatform-solr-search-engine:dev-master
    # (so that composer resolves ezpublish-kernel's self.version)
    git fetch --unshallow
fi

# Switch to another Symfony version if asked for
if [ "$SYMFONY_VERSION" != "" ] ; then
    composer require --no-update symfony/symfony="${SYMFONY_VERSION}"
    COMPOSER_UPDATE="true"
fi

# Install packages with composer update if asked for to make sure not use composer.lock if present
if [ "$COMPOSER_UPDATE" = "true" ] ; then
    echo "> Install dependencies through Composer (using update as other packages was requested)"
    composer update --no-progress --no-interaction --prefer-dist
else
    echo "> Install dependencies through Composer"
    composer install --no-progress --no-interaction --prefer-dist
fi
```

Comment (bdunogier) [inline bin/.travis/prepare_unittest.sh]: Better indeed. Didn't want to modify the file too much, but heck :-)

**Comment by bdunogier** [comment]
~~That's why. All we need to do is run `git fetch --unshallow` on the tested branch, and it should all work. Will try.~~

Nope, didn't work out. Doing a new try by checking out a branch named `6.1`. We'll see.

**Comment by bdunogier** [comment]
> Doing a new try by checking out a branch named 6.1. 

This one worked. Cleaned up the PR, ready to be merged.
